web/src/app/admin/AdminDialog.tsx

```diff
@@ -0,0 +1,111 @@
+import React from 'react'
+import List from '@mui/material/List'
+import ListItem from '@mui/material/ListItem'
+import ListItemText from '@mui/material/ListItemText'
+import Typography from '@mui/material/Typography'
+import { omit } from 'lodash'
+import FormDialog from '../dialogs/FormDialog'
+import { nonFieldErrors, fieldErrors } from '../util/errutil'
+import Diff from '../util/Diff'
+import {
+  DocumentNode,
+  OperationVariables,
+  TypedDocumentNode,
+  useMutation,
+} from '@apollo/client'
+
+interface Value {
+  id: string
+  oldValue: number
+  value: number
+  type: string
+}
+
+interface FieldValues {
+  [id: string]: string
+}
+
+interface AdminDialogProps {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  mutation: DocumentNode | TypedDocumentNode<any, OperationVariables>
```

Comment:
My IDE seems fine with using just `DocumentNode`, so we can remove the `any` reference

```suggestion
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  mutation: DocumentNode
```